internal/measurement/measurement.go

```diff
@@ -262,67 +249,125 @@ func timeLabel(value int64, fromUnit, toUnit string) (v float64, u string, ok bo
 		toUnit = strings.TrimSuffix(toUnit, "s")
 	}
 
-	var d time.Duration
-	switch fromUnit {
-	case "nanosecond", "ns":
-		d = time.Duration(value) * time.Nanosecond
-	case "microsecond":
-		d = time.Duration(value) * time.Microsecond
-	case "millisecond", "ms":
-		d = time.Duration(value) * time.Millisecond
-	case "second", "sec", "s":
-		d = time.Duration(value) * time.Second
-	case "cycle":
+	if fromUnit == "cycle" {
 		return float64(value), "", true
-	default:
+	}
+
+	uf := func(unit string) (string, float64, bool) {
+		return unitFactor(unit, timeUnits)
+	}
+	as := func(value float64) (float64, string, bool) {
+		return autoscale(value, timeUnits)
+	}
+	return convertUnit(value, fromUnit, toUnit, unit{"s", []string{}, float64(time.Second)}, uf, as)
+}
+
+var caseInsensitiveGCUUnits = []unit{
+	{"n·GCU", []string{"nanogcu"}, 1e-9},
+	{"μ·GCU", []string{"microgcu"}, 1e-6},
+	{"m·GCU", []string{"milligcu"}, 1e-3},
+	{"GCU", []string{"gcu"}, 1},
+	{"k·GCU", []string{"kilogcu"}, 1e3},
+	{"M·GCU", []string{"megagcu"}, 1e6},
+	{"G·GCU", []string{"gigagcu"}, 1e9},
+	{"T·GCU", []string{"teragcu"}, 1e12},
+	{"P·GCU", []string{"petagcu"}, 1e15},
+}
+
+var caseSensitiveGCUUnits = []unit{
+	{"n·GCU", []string{"n·GCU", "n*GCU"}, 1e-9},
+	{"μ·GCU", []string{"μ·GCU", "μ*GCU", "u·GCU", "u*GCU"}, 1e-6},
```

Comment:
I'm again a bit cautious about unicode characters in a CLI tool. Is it safe to assume that everybody's terminal is capable of printing these characters?

Reply:
Done (here and elsewhere)